aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsMetricAttributeGenerator.java

```diff
@@ -152,35 +212,132 @@ private static void setEgressOperation(SpanData span, AttributesBuilder builder)
    * and RPC attributes to use here, but this is a sufficient starting point.
    */
   private static void setRemoteServiceAndOperation(SpanData span, AttributesBuilder builder) {
```

Comment:
I thought we said that remote service/remote operation would be done independently (e.g. could have rpc service and db operation)

Reply:
it could happen in the following blocks which is not restricted by pair of values.

```
// try to derive RemoteService and RemoteOperation from the other related attributes
    if (remoteService.equals(UNKNOWN_REMOTE_SERVICE)) {
      remoteService = generateRemoteService(span);
    }
    if (remoteOperation.equals(UNKNOWN_REMOTE_OPERATION)) {
      remoteOperation = generateRemoteOperation(span);
    }
```

Reply:
Discussed offline - not exactly what I was talking about, but we decided this is desirable behaviour.